### DISCUSSION TITLE
Some weired issues about 1.18.2 worlds

### DISCUSSION THREAD

**Discussion by TheMeinerLP** [Minestom/Minestom#770] [Help]
Hello there, 
a friend(@theEvilReaper)  and me it self currently working at a self devloped rpg in minecraft(1.18.2). 
Yesterday(24.03.2022) we got a lobby for some testing stuff. 
We use the integrated anvil loader. 
But some chunks are not loading:
![2022-03-15_12 51 19](https://user-images.githubusercontent.com/6745190/158372110-70b48214-af26-464b-b699-1ad96e9cff23.png)
My next step was:
I explored the chunks wtih a NBTEditor to see whats wrong is. And found out the chunks can be have different version vs the level.dat(For me little bit new).
Beside we  got a stracktrace if the chunk loading:

> ```
> java.lang.IndexOutOfBoundsException: Index 9 out of bounds for length 5
>         at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
>         at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
>         at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
>         at java.base/java.util.Objects.checkIndex(Objects.java:359)
>         at java.base/java.util.ArrayList.get(ArrayList.java:427)
>         at org.jglrxavpok.hephaistos.mca.ChunkSection.<init>(ChunkSection.kt:158)
>         at org.jglrxavpok.hephaistos.mca.ChunkColumn.<init>(ChunkColumn.kt:205)
>         at org.jglrxavpok.hephaistos.mca.RegionFile.readColumn(RegionFile.kt:184)
>         at org.jglrxavpok.hephaistos.mca.RegionFile.getChunk$lambda-3(RegionFile.kt:110)
>         at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1740)
>         at org.jglrxavpok.hephaistos.mca.RegionFile.getChunk(RegionFile.kt:110)
>         at net.minestom.server.instance.AnvilLoader.loadMCA(AnvilLoader.java:80)
>         at net.minestom.server.instance.AnvilLoader.loadChunk(AnvilLoader.java:69)
>         at net.minestom.server.instance.InstanceContainer.lambda$retrieveChunk$8(InstanceContainer.java:260)
>         at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
>         at java.base/java.lang.Thread.run(Thread.java:833)
> ```

After that before i landing here, i search littlebit around in the source code. 
The index is out of the range of biomes.
Then i check it, how many biomes has the client and the `hephaistos` api.
For that i decompiled the client with fabric and was little bit confused. 
The api have 81 Biomes entries of the enum. But the client only 61...

> Also what intressting here:
> ```kt
> /**
>      * May not exist. If it does, it is 4*4*4 Strings of Biome IDs, for a 4x4x4 volume in the chunk (ie 1 string correspond
>      * to the biome for a cube of 4x4x4 blocks).
>      * Arranged by X, Z and then Y.
>      */
>     var biomes: Array<String>? = null
> 
>     private val biomeArraySize get()= 4*4*4
> ```

The limit of biomes are 64. But if the enum has entries 81, that can not be fit.

So the next question here is how can i fix the issue or what are the next steps to resolve the problem
 😄 

In pleasuer sorry for my bad english :(

**Comment by MrlnHi** [comment]
You should probably create an issue with the stack trace
